src/figuresClasses.ts

```diff
@@ -1,11 +1,92 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export abstract class Shape implements Figure {
+  // eslint-disable-next-line no-useless-constructor
+  constructor(public color: string) {}
+
+  abstract shape: string;
+  abstract getArea(): number;
+}
+
+export class Triangle extends Shape {
+  constructor(
+    color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    super(color);
+
+    if (a <= 0 || b <= 0 || c <= 0 || a + b <= c || a + b <= c || b + c <= a) {
+      throw new Error('Sides should be positive and form a valid triangle.');
+    }
+    this.shape = 'triangle';
+  }
+
+  shape: string;
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+
+    return Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+  }
+}
 
-export class Triangle implements Figure {}
+export class Circle extends Shape {
+  constructor(
+    color: string,
+    public radius: number,
+  ) {
+    super(color);
 
-export class Circle implements Figure {}
+    if (radius <= 0) {
+      throw new Error('Radius should be a positive number');
+    }
+    this.shape = 'circle';
+  }
+
+  shape: string;
+
+  getArea(): number {
+    const square: number = Math.PI * Math.pow(this.radius, 2);
+
+    return Math.floor(square * 100) / 100;
+  }
+}
+
+export class Rectangle extends Shape {
+  constructor(
+    color: string,
+    public width: number,
+    public height: number,
+  ) {
+    super(color);
+
+    if (width <= 0 || height <= 0) {
+      throw new Error('Width and Height should be positive!');
+    }
+    this.shape = 'rectangle';
+  }
+
+  shape: string;
+
+  getArea(): number {
+    const square: number = this.height * this.width;
+
+    return square;
+  }
+}
 
-export class Rectangle implements Figure {}
+export function getInfo(figure: Figure): string {
+  const area: number = figure.getArea();
 
-export function getInfo(figure): string {
-  return typeof figure;
+  if (Number.isInteger(area)) {
+    return `A ${figure.color} ${figure.shape} - ${area}`;
+  } else {
```

Comment:
Else is not needed here, because the **if** contains a return. So, when the if runs, nothing else will run after.
In this way, you could move the return inside the **else** to the function itself as a default return.

Another way possible would be to simply return the string and use a **ternary operator** for the last parameter inside of it. like so:
```
 return `A ${figure.color} ${figure.shape} - ${Number.isInteger(area) ? area : area.toFixed(2)}`; 
```
Or you could always use the method .toFixed(2) like so: (if the number is an integer, it will return the number with two zeros as decimal cases)
```
 return `A ${figure.color} ${figure.shape} - ${area.toFixed(2)}`; 
```

And last, but not least, *getInfo* could be a method of the *Shape* class directly and access all needed values using *this*. But no problem having it here.

Reply:
That approach is way cleaner, thank's!

I tried to set _getInfo_ as a method of _Shape_ , but this change fails in some tests, so I'm keeping the function apart from the class if that's okay..

As what I can understand, I can't set this method inside the class, 'cause I need to call it and pass the object as argument, is that right? Or am I able to do it but and I'm missing something?

"Also create a function `getInfo` that takes a figure and returns a string in the
next format:"
```typescript
const redRectangle = new Rectangle('red', 3, 5);
getInfo(redRectangle) === 'A red rectangle - 15';

const greenCircle = new Circle('green', 1);
getInfo(greenCircle) === 'A green circle - 3.14';
```